frontend/app/components/work-packages/wp-attachments-formattable-field/wp-attachments-formattable.directive.ts

```diff
@@ -0,0 +1,131 @@
+                                }
+                                else if (dropData.filesCount > 1) {
+                                    for (let i: number = updatedAttachments.length - 1;
+                                      i >= updatedAttachments.length - dropData.filesCount;
+                                      i--) {
+                                        description.insertAttachmentLink(
+                                          updatedAttachments[i]._links.downloadLocation.href,
+                                          InsertMode.ATTACHMENT,
+                                          true);
+                                  }
+                                }
+                                description.save();
+                            },(err)=>{
+                              console.log("error while reloading attachments",err)
+                            });
+                    }, function(err) {
+                        console.log(err);
+                    });
+                }
+                else {
+                  dropData.files.forEach((file: File) => {
+                    description.insertAttachmentLink(file.name.replace(/ /g , '_'), InsertMode.ATTACHMENT, true);
+                    file['isPending'] = true;
+                    this.wpAttachments.addPendingAttachments(file);
+                  });
+                  description.save();
+                }
+            }
+        }
+        else {
+            const insertUrl: string = dropData.isAttachmentOfCurrentWp() ? dropData.removeHostInformationFromUrl() : dropData.webLinkUrl;
+            const insertAlternative: InsertMode = dropData.isWebImage() ? InsertMode.INLINE : InsertMode.LINK;
+            const insertMode: InsertMode = dropData.isAttachmentOfCurrentWp() ? InsertMode.ATTACHMENT : insertAlternative;
+
+            description.insertWebLink(insertUrl, insertMode);
+            description.save();
+        }
+    };
+
+    protected prevDefault(evt: DragEvent): void {
+        evt.preventDefault();
+        evt.stopPropagation();
+    }
+}
+
+function wpAttachmentsFormattable() {
+    return {
+      bindToController: true,
+      controller: WpAttachmentsFormattableController,
+      link: function(scope: ng.IScope, 
+                     element: ng.IAugmentedJQuery, 
+                     attrs: ng.IAttributes, 
+                     controllers: Array<ng.IControllerService>){
+        // right now the attachments directive will only work in combination with either
+        // the wpSingleView or the wpEditForm directive
+        // else the drop handler will fail because of a missing reference to the current wp
+        if(angular.isUndefined(controllers[0] && angular.isUndefined(controllers[1])))
+          return;
+
+        scope.workPackage = (controllers[0] === null) ? controllers[1].workPackage : controllers[0].workPackage;
```

Comment:
I wonder why this line would be necessary. The wpEditFormController should always have the work package anyway.